### ISSUE TITLE
@aws-sdk/client-s3 DeleteObjects command  is breaking due to inclusion of x-id query param 

### ISSUE THREAD

**Issue by arciisine** [localstack/localstack#3931]
# Type of request
[x] bug report

# Detailed description
This seems to be a change that was released on 4/14 in the aws-sdk-js-v3 [Line](https://github.com/aws/aws-sdk-js-v3/blame/main/clients/client-s3/protocols/Aws_restXml.ts#L1356).

By modifying the aws-sdk to not send `x-id=DeleteObjects`, the code works perfectly fine.  Leaving it in, causes the request to break, and return a 500.

I see some code in [s3_listener.py](https://github.com/localstack/localstack/blob/9552b4b994917d61d47823b6663b3b7b6fd34cfa/localstack/services/s3/s3_listener.py#L962) that references `x-id`, but not for this operation.  I'm not sure how it all plays together, but it looks like the change in the sdk is having unintended consequences.

## Expected behavior
Delete objects should work as expected, without issue.

## Actual behavior
Delete objects is failing with a 500, due to the presence of the `x-id` query parameter. 

# Steps to reproduce

Use the latest @aws-sdk/client-s3 module, and issue a deleteObjects operation.  

## Command used to start LocalStack
```bash
docker run --privileged=true \
  -p 4566 -p 4571 -p 8080 -p 8081 \
  -e TEST_AWS_ACCOUNT_ID=000000000000 \
  -e LOCALSTACK_HOSTNAME=localhost \
  -e DEFAULT_REGION=us-east-1 \
  -e HOST_TMP_FOLDER=/tmp/folder \
  -v /tmp/folder:/tmp/localstack \
  localstack/localstack:0.12.10`
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
```typescript
await this.client.deleteObjects({
  Bucket: this.config.bucket,
  Delete: {
    Objects: items // [{ Key: 'key1'}, {Key: 'key2'}...]
  }
});
```

**Comment by pandomic** [comment]
Thanks for reporting @arciisine. We will keep you informed (or in case we need any more details)

**Comment by pandomic** [comment]
@arciisine I was unable to reproduce the issue using `Localstack@0.12.10` & `@aws-sdk/client-s3@3.13.1`.

package.json
```json
{
  "name": "sample-repo",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@aws-sdk/client-s3": "^3.13.1"
  }
}

```

test.js
```js
// @aws-sdk/client-s3 @ 3.13.1
const {
    S3Client,
    CreateBucketCommand,
    PutObjectCommand,
    ListObjectsCommand,
    DeleteObjectCommand
} = require("@aws-sdk/client-s3");

const BUCKET_NAME = 'sample-bucket-8';
const FILE_NAME = 'test.txt';
const FILE_CONTENTS = 'Hello World!'

const client = new S3Client({
    endpoint: 'http://s3.localhost.localstack.cloud:4566',
    region: 'us-east-1'
});

(async () => {
    // 1. Create Bucket

    const createBucketCommand = new CreateBucketCommand({
        Bucket: BUCKET_NAME
    });

    await client.send(createBucketCommand);

    // 2. Put sample file

    const putObjectCommand = new PutObjectCommand({
        Bucket: BUCKET_NAME,
        Key: FILE_NAME,
        Body: FILE_CONTENTS
    });

    await client.send(putObjectCommand);

    // 3. Make sure it's there

    const listObjectsCommand = new ListObjectsCommand({
        Bucket: BUCKET_NAME
    });

    const listResponse = await client.send(listObjectsCommand);
    console.info('Before deletion', (listResponse['Contents'] || []).map(({ Key }) => Key));

    // 4. Testing deletion

    const deleteObjectsCommand = new DeleteObjectCommand({
        Bucket: BUCKET_NAME,
        Key: FILE_NAME
    });

    await client.send(deleteObjectsCommand);

    // 3. Make sure it's gone

    const listResponseAfterDeletion = await client.send(listObjectsCommand);
    console.info('After deletion', (listResponseAfterDeletion['Contents'] || []).map(({ Key }) => Key));
})();
```

I also noticed a couple of issues with your report:
* You haven't forwarded port to your host machine (`-p 4566:4566`)
* Provided code snippet does not look like v3 API

Could you please double-check your issue and let us know if it worked?

**Comment by arciisine** [comment]
Just to clarify, my local scenario is slightly more complicated, and hard to isolate, so my docker command may have missed a few parameters.

Looking at your example, it isn't using `DeleteObjectsCommand`, but is using `DeleteObjectCommand` . The issue is specifically with the `DeleteObjectsCommand` as it produces a different `http` request.

**Comment by pandomic** [comment]
Thank you for clarifications @arciisine, I can confirm this is a **bug**. We already identified the problem and prepared a potential fix. Stay tuned for an upcoming release.

**Comment by whummer** [comment]
Great, thanks @arciisine and @pandomic . This should be fixed in #3937 - can you please pull the `latest` Docker image and give it another try? Please let us know if the problem persists. Thanks!

**Comment by arciisine** [comment]
@whummer I've pulled the latest image (twice over the course of a few days), and I'm still seeing the same issue.  Is there anything I should be looking for in the image to help verify that this is resolved, and that I'm actually running with the appropriate code?

**Comment by whummer** [comment]
Thanks for the update @arciisine , we'll look into this issue again. Just to confirm - this is still happening with the same client code as reported initially, right? 
```
await this.client.deleteObjects({
  Bucket: this.config.bucket,
  Delete: {
    Objects: items // [{ Key: 'key1'}, {Key: 'key2'}...]
  }
});
```
/cc @pandomic

**Comment by whummer** [comment]
Just to confirm again @arciisine - please make sure the use the `localstack/localstack:latest` (not the tagged version) in your startup command:
```
$ docker pull localstack/localstack
$ docker run --privileged=true \
  -p 4566 -p 4571 -p 8080 -p 8081 \
  -e TEST_AWS_ACCOUNT_ID=000000000000 \
  -e LOCALSTACK_HOSTNAME=localhost \
  -e DEFAULT_REGION=us-east-1 \
  -e HOST_TMP_FOLDER=/tmp/folder \
  -v /tmp/folder:/tmp/localstack \
  localstack/localstack:latest
```

Can you please share the version info printed in the startup logs of your LocalStack container:
```
LocalStack version: 0.12.10
LocalStack Docker container id: 6274182a747d
LocalStack build date: 2021-05-02
LocalStack build git hash: 481a540b
```
Thanks

**Comment by arciisine** [comment]
I'll verify.  I checked the source in the running container, and it had the
code changes, but I'll verify the container information later today.

_tim

**Comment by arciisine** [comment]
@whummer 

This is the latest I have.  I noticed the date is different, but 5/1 is when I ran the tests.

```
Starting local dev environment. CTRL-C to quit.

LocalStack version: 0.12.10
LocalStack build date: 2021-05-01
LocalStack build git hash: 2614a54c

Starting edge router (https port 4566)...
```

**Comment by arciisine** [comment]
Just tried again (still failling locally), with:
```
2021-05-06 15:37:30,711 INFO success: infra entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
Waiting for all LocalStack services to be ready

LocalStack version: 0.12.11
LocalStack build date: 2021-05-06
LocalStack build git hash: c8084846

```

**Comment by pandomic** [comment]
Hi @arciisine. Unfortunately, I was unable to reproduce the issue using the latest localstack image (build 2021-05-16) + `@aws-sdk/client-s3@3.16.0` and the following snippet (modified version from above):

```js
// @aws-sdk/client-s3 @ 3.13.1
const {
  S3Client,
  CreateBucketCommand,
  PutObjectCommand,
  ListObjectsCommand,
  DeleteObjectsCommand
} = require("@aws-sdk/client-s3");

const BUCKET_NAME = 'sample-bucket-8';
const FILE_NAME = 'test.txt';
const FILE_CONTENTS = 'Hello World!'

const client = new S3Client({
  endpoint: 'http://s3.localhost.localstack.cloud:4566',
  region: 'us-east-1'
});

(async () => {
  // 1. Create Bucket

  const createBucketCommand = new CreateBucketCommand({
      Bucket: BUCKET_NAME
  });

  await client.send(createBucketCommand);

  // 2. Put sample file

  const putObjectCommand = new PutObjectCommand({
      Bucket: BUCKET_NAME,
      Key: FILE_NAME,
      Body: FILE_CONTENTS
  });

  await client.send(putObjectCommand);

  // 3. Make sure it's there

  const listObjectsCommand = new ListObjectsCommand({
      Bucket: BUCKET_NAME
  });

  const listResponse = await client.send(listObjectsCommand);
  console.info('Before deletion', (listResponse['Contents'] || []).map(({ Key }) => Key));

  // 4. Testing deletion

  const deleteObjectsCommand = new DeleteObjectsCommand({
      Bucket: BUCKET_NAME,
      Delete: { Objects: [{ Key: FILE_NAME }] }
  });

  await client.send(deleteObjectsCommand);

  // 3. Make sure it's gone

  const listResponseAfterDeletion = await client.send(listObjectsCommand);
  console.info('After deletion', (listResponseAfterDeletion['Contents'] || []).map(({ Key }) => Key));
})();
```

The output I get:
![image](https://user-images.githubusercontent.com/7187922/118415128-47807680-b6b1-11eb-95bd-d4d1e357354f.png)

Could you provide a sample repository that reproduces the error, this will let us help a way faster. Thank you.

**Comment by whummer** [comment]
Thanks @arciisine . This should now (hopefully) be fixed in #4140 - can you please pull the `latest` Docker image and give it another try? Please let us know if the problem persists. Thanks!
